libvast/src/format/syslog.cpp

```diff
@@ -103,13 +110,14 @@ reader::read_impl(size_t max_events, size_t max_slice_size, consumer& f) {
       return false;
     } else {
       return lines_->next_timeout(
-        vast::defaults::import::shared::partial_slice_read_timeout);
+        std::chrono::duration_cast<std::chrono::milliseconds>(read_timeout_));
```

Comment:
We can get rid of all these duration_casts by making `read_timeout_` be `std::chrono::milliseconds` instead.

(I didn't verify this, but in theory casting from nanoseconds to milliseconds should involve a division by 10000, so it's probably not free to do this cast)

Reply:
I'd rather change the interface of `next_timeout` to work with every kind of clock as would be expected.

Reply:
It's not the clock that is the problem, but the duration that the passed number represents. At some point `line_range` has to pass an integer representing milliseconds to `poll()`, so if you pass in something different there is a cost that needs to paid for converting.

Reply:
It doesn't really matter where we do the conversion: It's either the result of `steady_clock::now()` or the parameter for `next_timeout`.

I've added an overload to `next_timeout` that does the conversion for us so that we can use it with every `std::chrono::duration<Rep, Period>`.